File: sample-chat-java/src/main/java/com/quickblox/sample/chat/java/ui/adapter/UsersAdapter.java
package com.quickblox.sample.chat.java.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;

import com.quickblox.chat.QBChatService;
import com.quickblox.sample.chat.java.R;
import com.quickblox.sample.chat.java.utils.ResourceUtils;
import com.quickblox.sample.chat.java.utils.UiUtils;
import com.quickblox.users.model.QBUser;

import java.util.List;

public class UsersAdapter extends BaseAdapter {

    protected List<QBUser> userList;
    protected QBUser currentUser;
    private Context context;

    public UsersAdapter(Context context, List<QBUser> users) {
        this.context = context;
        currentUser = QBChatService.getInstance().getUser();
        userList = users;
        addCurrentUserToUserList();
    }

    private void addCurrentUserToUserList() {
        if (currentUser != null) {
            if (!userList.contains(currentUser)) {
                userList.add(currentUser);
            }
        }
    }

    public void addUserToUserList(QBUser user) {
        if (!userList.contains(user)) {
            userList.add(user);
        }
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        QBUser user = getItem(position);

        ViewHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(context).inflate(R.layout.list_item_user, parent, false);
            holder = new ViewHolder();
            holder.userImageView = (ImageView) convertView.findViewById(R.id.image_user);
            holder.loginTextView = (TextView) convertView.findViewById(R.id.text_user_login);
            holder.userCheckBox = (CheckBox) convertView.findViewById(R.id.checkbox_user);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        if (isUserMe(user)) {
            holder.loginTextView.setText(context.getString(R.string.placeholder_username_you, user.getFullName()));
        } else {
            holder.loginTextView.setText(user.getFullName());
        }

        if (isAvailableForSelection(user)) {
            holder.loginTextView.setTextColor(ResourceUtils.getColor(R.color.text_color_black));
        } else {
            holder.loginTextView.setTextColor(ResourceUtils.getColor(R.color.text_color_medium_grey));
        }

        holder.userImageView.setBackgroundDrawable(UiUtils.getColorCircleDrawable(position));
        holder.userCheckBox.setVisibility(View.GONE);

        return convertView;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getCount() {
        return userList.size();
    }

    @Override
    public QBUser getItem(int position) {
        return userList.get(position);
    }

    private boolean isUserMe(QBUser user) {
        return currentUser != null && currentUser.getId().equals(user.getId());
    }

    protected boolean isAvailableForSelection(QBUser user) {
        return currentUser == null || !currentUser.getId().equals(user.getId());
    }

    protected static class ViewHolder {
        ImageView userImageView;
        TextView loginTextView;
        CheckBox userCheckBox;
    }
}

File: sample-chat-kotlin/app/src/main/java/com/quickblox/sample/chat/kotlin/ui/adapter/UsersAdapter.kt
package com.quickblox.sample.chat.kotlin.ui.adapter

import android.content.Context
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import com.quickblox.chat.QBChatService
import com.quickblox.sample.chat.kotlin.R
import com.quickblox.sample.chat.kotlin.utils.getColorCircleDrawable
import com.quickblox.users.model.QBUser


open class UsersAdapter(val context: Context, val userList: MutableList<QBUser>) : BaseAdapter() {
    private var currentUser: QBUser? = QBChatService.getInstance().user

    fun addNewList(users: List<QBUser>) {
        userList.clear()
        userList.addAll(users)

        for (user in users) {
            if (isUserMe(user)) {
                userList.remove(user)
            }
        }
        notifyDataSetChanged()
    }

    fun addUsers(users: List<QBUser>) {
        users.forEach {
            if (!userList.contains(it)) {
                userList.add(it)
            }
        }
        notifyDataSetChanged()
    }

    fun removeUsers(users: List<QBUser>) {
        for (user in users) {
            userList.remove(user)
        }
        notifyDataSetChanged()
    }

    fun clearList() {
        userList.clear()
        notifyDataSetChanged()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val holder: ViewHolder
        var modifiedView = convertView
        val user = getItem(position)

        if (modifiedView == null) {
            modifiedView = LayoutInflater.from(context).inflate(R.layout.list_item_user, parent, false) as View
            holder = ViewHolder()
            holder.rootLayout = modifiedView.findViewById(R.id.item_root_layout)
            holder.userImageView = modifiedView.findViewById(R.id.image_user)
            holder.loginTextView = modifiedView.findViewById(R.id.text_user_login)
            holder.userCheckBox = modifiedView.findViewById(R.id.checkbox_user)
            holder.userAvatarTitle = modifiedView.findViewById(R.id.text_user_avatar_title)
            modifiedView.tag = holder
        } else {
            holder = modifiedView.tag as ViewHolder
        }

        val username = if (TextUtils.isEmpty(user.fullName)) {
            user.login
        } else {
            user.fullName
        }

        if (isUserMe(user)) {
            holder.loginTextView.text = context.getString(R.string.placeholder_username_you, username)
        } else {
            holder.loginTextView.text = username
        }

        if (isAvailableForSelection(user)) {
            holder.loginTextView.setTextColor(context.resources.getColor(R.color.text_color_black))
        } else {
            holder.loginTextView.setTextColor(context.resources.getColor(R.color.text_color_medium_grey))
        }

        holder.userImageView.setBackgroundDrawable(getColorCircleDrawable(user.id.hashCode()))
        holder.userCheckBox.visibility = View.GONE

        if (!TextUtils.isEmpty(user.fullName)) {
            val avatarTitle = user.fullName.get(0).toString().toUpperCase()
            holder.userAvatarTitle.text = avatarTitle
        }

        return modifiedView
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getCount(): Int {
        return userList.size
    }

    override fun getItem(position: Int): QBUser {
        return userList[position]
    }

    private fun isUserMe(user: QBUser): Boolean {
        return currentUser?.id == user.id
    }

    protected open fun isAvailableForSelection(user: QBUser): Boolean {
        return currentUser?.id != user.id
    }

    protected class ViewHolder {
        lateinit var rootLayout: LinearLayout
        lateinit var userImageView: ImageView
        lateinit var loginTextView: TextView
        lateinit var userCheckBox: CheckBox
        lateinit var userAvatarTitle: TextView
    }
}